src/helpers/serverModuleHelpers.js

```diff
@@ -0,0 +1,118 @@
+const { existsSync } = require('node:fs')
+const { readFile, writeFile, rename, rm } = require('node:fs/promises')
+const { parse, join } = require('node:path')
+
+const { satisfies } = require('semver')
+
+const getAngularJson = require('./getAngularJson')
+const { getProject } = require('./setUpEdgeFunction')
+
+// eslint-disable-next-line no-inline-comments
+const NetlifyServerTsCommonEngine = /* typescript */ `import { CommonEngine } from '@angular/ssr/node'
+import { render } from '@netlify/angular-runtime/common-engine'
+
+const commonEngine = new CommonEngine()
+
+export default async function HttpHandler(request: Request, context: any): Promise<Response> {
+  return await render(commonEngine)
+}
+`
+
+// eslint-disable-next-line no-inline-comments
+const NetlifyServerTsAppEngine = /* typescript */ `import { AngularAppEngine, createRequestHandler } from '@angular/ssr'
+const angularAppEngine = new AngularAppEngine()
+
+// @ts-expect-error - createRequestHandler expects a function with single Request argument and doesn't allow context argument
```

Comment:
Possibly this `@ts-expect-error` directive will be no longer needed IF https://github.com/angular/angular-cli/pull/28845 would be accepted

Reply:
Types are checked at `ng build` time and it would fail otherwise - see https://github.com/netlify/angular-runtime/actions/runs/11798216865/job/32863885795#step:6:1537 for example before I (re)added that `@ts-expect-error` directive.

There are options to change that ( https://github.com/angular/angular-cli/issues/19948 ) but that's not the default.

There are some ways to maybe do things differently:
 - `reqHandler` named export is the one that would be used for dev server - and this one NEED to use `createRequestHandler` wrapped handler - given dev we wouldn't be getting context, so maybe this one could use use `request`
 - use other export for prod handler in which case we don't need to use `createRequestHandler` wrapping and no types to adhere to

Reply:
Ok, I think I figured workaround - we can grab context from https://docs.netlify.com/edge-functions/api/#netlify-global-object and then we should be able to skip the type problem here by just not getting context from 2nd argument for edge functions and only grab request (1st param) which would be compatible with Angular's `createRequestHandler` without adjusting type like in my optimistic PR to `@angular/ssr`

Reply:
I did end up making it work without those `@ts-expect-error` directives with usage of global Netlify object, but it still is not the best as `context` is typed as `any` because I can't use actual recent `@netlify/edge-functions` (that contain context on global object) because they reference `Deno` global types which Angular compiler doesn't know about those - users could install Deno types maybe to workaround this, but it does feel like some changes need to happen in `@netlify/edge-functions` to make it easy to use in Angular which doesn't typecheck in ~Deno runtime. Additionally to make it work I did add new module with `getContext()` export that basically just returns Netlify.context from global object, because directly trying to grab it from global Netlify was resulting in another type error in Angular land.

It generally seem like we either have to use `any` or we vendor `NetlifyContext` type (to not rely on `@netlify/edge-functions` package which is not usable in Angular context since `@netlify/edge-functions@2.10.0` which added relience on global Deno types)